### ISSUE TITLE
App crashing continously

### ISSUE THREAD

**Issue by Hemant0601** [QuickBlox/quickblox-android-sdk#329]
My app crash because of the following stack trace, I cant even handle it on my own as its inside the jar file.
SDK 3.0
No- specific environments, coming randomly for random devices and random users.

Fatal Exception: com.quickblox.core.exception.QBRuntimeException: Incorrect presence type
       at com.quickblox.chat.model.QBPresence$Type.fromSmackPresenceType(SourceFile:235)
       at com.quickblox.chat.model.QBPresence.(SourceFile:68)
       at com.quickblox.chat.QBGroupChat$ParticipantListener$1.run(SourceFile:548)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7229)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)

Please check.

**Comment by RomanPronin** [comment]
Hi @Hemant0601 , could you tell us more information about this issue. You use Privacy list, so you have blocked users, or maybe anything else?

**Comment by Hemant0601** [comment]
@RomanPronin Thanks, I am using Privacy list, any idea to ignore those users to send presence.

**Comment by RomanPronin** [comment]
For example, to detect blocked user you can use next snippet:
```
public boolean isUserBlocked(String privacyListName, String userID) {
        QBPrivacyListsManager privacyListsManager = QBChatService.getInstance().getPrivacyListsManager();

        List<QBPrivacyList> lists = null;
        try {
            lists = privacyListsManager.getPrivacyLists();
            for (QBPrivacyList qbPrivacyList : lists) {
                if (qbPrivacyList.getName().equals(privacyListName)) {

                    for (QBPrivacyListItem item : qbPrivacyList.getItems()) {
                        if (item.getValueForType().contains(userID)) {
                            return true;
                        }
                    }
                }
            }
        } catch (SmackException.NotConnectedException | SmackException.NoResponseException | XMPPException.XMPPErrorException e) {
            e.printStackTrace();
        }
        return false;
    }
```
and if user blocked then do not create roster createEntry with him:
```
    private void createRosterEntry()  {
        String privacyListName = "listName";
        String userID = "125";
        if (isUserBlocked(privacyListName, userID)) {
            Log.d(TAG(), "User is Blocked");
        } else {
            roster.createEntry(Integer.valueOf(userID), null);
        }
    }
```

**Comment by Hemant0601** [comment]
@RomanPronin Thanks for the snippet, I understood this, but what about once roster entry is created, and after all how can i update the other users Roster Entry. Eg, If User A blocks user B, then user A can remove him from his own Roster, but the issue is User B is sending his presence to user A, because user A still exists in user B's roster. Please explain.

**Comment by RomanPronin** [comment]
So, you have to remove B rosterEntry too -  roster.removeEntry(entry);  or make chatRoster.unsubscribe(userID); Actually we are working on this problem with Incorrect presence type and RuntimeException.

**Comment by Hemant0601** [comment]
@RomanPronin Thanks if you are working on , just let me know the new patch when its done. But the above suggestion will do for A only , B will have no idea about that and cant do that for person B.

**Comment by RomanPronin** [comment]
Fixed in the SDK version 3.3.0
